include/solvers/time_solver.h

```diff
@@ -261,12 +280,33 @@ class TimeSolver : public ReferenceCountedObject<TimeSolver>
   { _is_adjoint = _is_adjoint_value; }
 
   /**
-   * Function to return 'last_deltat()', returns system.deltat if
-   * fixed timestep solver is used, completed_deltat if the adaptive
+   * Returns system.deltat if fixed timestep solver is used,
+   * the complete timestep size (sum of all substeps) if the adaptive
    * time solver is used.
    */
-  virtual Real last_complete_deltat();
+  virtual Real last_completed_timestep_size();
+
+  /**
+   * Returns the system.deltat timestep used to get to the current time.
+   * For a uniform timestepper, it will simply return the last used system.deltat, for an
+   * adaptive time solver, it will return the last used substep size.
+   */
+  virtual Real last_completed_deltat();
 
+  /**
+   * The final_time variable can be set by the user or in another library
+   * function. By default the final_time variable is set to be std::nan("1").
```

Comment:
Nobody cares about the default setting value for it's own sake what's important is the default *behavior*, and how to restore that if necessary.

Although... is this a safe default value?  C++ doesn't technically guarantee that an implementation supports quiet NaN, and we don't want to throw a SIGFPE if someone has "scream if we ever see NaN" settings enabled somehow.  If the only thing we really *need* this for is to handle delta function QoIs at the final_time then couldn't we just default it to ```numeric_limits<Real>::max()```?

Reply:
Done.